platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -108,108 +122,140 @@ public Properties loadProperties(String propertiesFile) {
    * @return Properties deserialized from the specified file, or empty if the load failed
    */
   public Properties loadProperties(String propertiesFile, ClassLoader classLoader) {
-    boolean error = false;
     Properties properties = new Properties();
-    if (propertiesFile != null) {
-      try {
-        LOGGER.debug(
-            "Attempting to load properties from {} with Spring PropertiesLoaderUtils.",
-            propertiesFile);
-        properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile);
-      } catch (IOException e) {
-        error = true;
-        LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
-      }
-      if (error || properties.isEmpty()) {
-        if (classLoader != null) {
-          try {
-            LOGGER.debug(
-                "Attempting to load properties from {} with Spring PropertiesLoaderUtils with class loader.",
-                propertiesFile);
-            properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile, classLoader);
-            error = false;
-          } catch (IOException e) {
-            error = true;
-            LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
-          }
-        } else {
-          try {
-            LOGGER.debug(
-                "Attempting to load properties from {} with Spring PropertiesLoaderUtils with class loader.",
-                propertiesFile);
-            properties =
-                PropertiesLoaderUtils.loadAllProperties(
-                    propertiesFile, PropertiesLoader.class.getClassLoader());
-            error = false;
-          } catch (IOException e) {
-            error = true;
-            LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
-          }
-        }
+    if (propertiesFile == null) {
+      LOGGER.debug("Properties file must not be null.");
+      return properties;
+    }
+
+    Iterator<BiFunction<String, ClassLoader, Properties>> strategiesIterator =
+        PROPERTY_LOADING_STRATEGIES.iterator();
+    do {
```

Comment:
✏️ An alternative (neither better nor worse) would be something like:
```java
return PROPERTY_LOADING_STRATEGIES.stream()
  .map(f -> f.apply(propertiesFile, classLoader)
  .filter(Objects::nonNull)
  .filter(p -> !properties.isEmpty())
  .map(PropertiesLoader::substituteSystemPropertyPlaceholders);
```

Reply:
I'm personally too worried to make that change and have it behave exactly the same as it is now. I don't want some weird stream thing to bite me in the behind. I like the idea of using streams though.

Reply:
Clarifying point: we do NOT lose the benefit of short-circuiting with streams as they are lazily processed. Toss in the missing `findFirst()` after the second predicate `filter` (Not sure why I forgot it) and you'll stop processing on the first non-empty.